src/device/router_device_routing.erl

```diff
@@ -1185,80 +1182,66 @@ get_and_sort_devices(DevAddr, PubKeyBin) ->
 get_device_by_mic(_MIC, _Payload, []) ->
     undefined;
 get_device_by_mic(ExpectedMIC, Payload, [Device | Devices]) ->
-    ExpectedFCnt = expected_fcnt(Device, Payload),
```

Comment:
Don't guess at what the fcnt should be. 
Iterate through the higher frame bits and determine for certain what the frame count is. 
Then we can pass that fcnt through to the device worker and know it's verified.